### PR TITLE
Fix #1467: Display a QuickLook preview when tapping a downloaded file

### LINKED ISSUE #1467: Need a way to view download content
New download feature has no way to view download. Downloaded a jpg and wasn’t able to open for viewing. 

Would be great to use QL Preview controller here as it provides lots of native functionality and supports many different file types.

### PR DESCRIPTION
**Pull request by kylehickinson**
Falls back to previous logic if the QuickLook service isn't able to preview the document type

## Summary of Changes

This pull request fixes issue #1467 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

- Download a PDF, and maybe some images (also can just place them into the app's Download folder from Files app)
- Open Downloads menu and click on said documents
- Verify that the correct preview appears

## Screenshots:

Examples:

| PDF | Image |
| --- | --- |
| IMG_4D25B1111F18-2 | IMG_DC9D8E6F03C3-1 |

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/DownloadsViewController.swift [hunk 1]
```diff
@@ -295,13 +296,6 @@ class DownloadsPanel: UIViewController, UITableViewDelegate, UITableViewDataSour
         return configureDownloadedFile(cell, for: indexPath)
     }
     
-    func tableView(_ tableView: UITableView, willDisplayHeaderView view: UIView, forSection section: Int) {
-        if let header = view as? UITableViewHeaderFooterView {
-            header.textLabel?.textColor = UIColor.Photon.Grey90
-            header.contentView.backgroundColor = UIColor.Photon.Grey10
-        }
-    }
```
Comment (kylehickinson) [inline Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/DownloadsViewController.swift:303]: @iccub There was a bug that was messing up the text/background color when dismissing the preview. Let me know if this code is still needed for something else

Comment (iccub) [reply Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/DownloadsViewController.swift:303]: It's ok to remove, it was a copy pasta from FF

File: Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/DownloadsViewController.swift [hunk 2]
```diff
@@ -395,3 +400,25 @@ extension DownloadsPanel: Themeable {
...
     }
 }
+
+extension DownloadsPanel: QLPreviewControllerDelegate {
+    func previewControllerDidDismiss(_ controller: QLPreviewController) {
+        previewingFileDataSource = nil
+    }
+}
+
+class PreviewingDataSource: NSObject, QLPreviewControllerDataSource {
```
Comment (iccub) [inline Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/DownloadsViewController.swift]: Make private

**Review by iccub — CHANGES_REQUESTED** [review]
One code change + few other concerns shared on slack

**Review by iccub — APPROVED** [review]

**Pull request merged**